### PR TITLE
Ledgersmb admin user

### PR DESCRIPTION
**Pull request by ylavoie**
This add user management to `ledgersmb-admin`, allowing to list, create or change users.

### CODE REVIEW

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+
+use Moose;
+extends 'LedgerSMB::Admin::Command';
+use namespace::autoclean;
+
+use Feature::Compat::Try;
+
+# LedgerSMB database interface
+has db => (is => 'rw');
+has super_dbh => (is => 'rw',
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:30]: Not used?

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+                  predicate => '_has_super_dbh');
+has admin_dbh => (is => 'rw', lazy => 1,
+                  builder => '_build_admin_dbh',
+                  clearer => '_clear_admin_dbh',
+                  predicate => '_has_admin_dbh',
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:35]: The builder isn't defined.Do you really eed the admin_dbh, or isn't a remnant of an early version?

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+    my $_username = shift // $username;
+
+    # The code below doesn't provide the entity_id
+    #my $user_obj = LedgerSMB::DBObject::User->new();
+    #$user_obj->set_dbh($dbh);
+    #my @users = @{$user_obj->get_all_users};
+    #my ($user) = grep { $_username eq $_->{username} } @users;
+
+    #This ugly hack does
+    my $sth = $dbh->prepare("SELECT id, entity_id FROM users WHERE username='$username'");
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:89]: Please use placeholders and not string interpolation with SQL commands.

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+    if (!$user) {
+        $self->logger->error('Invalid user');
+        $dbh->rollback;
+        return 1;
+    }
+    #TODO: Fix validity
+    $dbh->do(qq(ALTER USER "$username" VALID UNTIL 'infinity'));
+
+    # Add permissions
+    return 1 if $permission && !$self->_add_permissions($dbh, $user);
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:295]: This prevents adding and removing permissions in a single invocation. Is that intentional?

**Review by ehuelsmann — CHANGES_REQUESTED** [review]
Yep. This one is long over due.... Thanks for taking the time to do it!

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 1 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm:30]: Remant. Removed

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 4 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm:295]: No, this fails early when permissions were given but aren't valid.
Same for no-permissions

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 2 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm:35]: Remnant. Removed

File: lib/LedgerSMB/Admin/Command/user.pm [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+use LedgerSMB::PGDate;
+
+use Moose;
+extends 'LedgerSMB::Admin::Command';
+use namespace::autoclean;
+
+use Feature::Compat::Try;
+
+# LedgerSMB database interface
+has db => (is => 'rw');
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Admin/Command/user.pm:29]: Are you sure this one is correct? Shouldn't it inherit it from `LedgerSMB::Admin::Command`?

Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm:29]: Indeed. Fixed.

File: lib/LedgerSMB/Admin/Command/user.pm (hunk 3 above)
Comment (ylavoie) [reply lib/LedgerSMB/Admin/Command/user.pm]: Done wherever possible.

**Review by ehuelsmann — APPROVED** [review]

**Pull request merged**